### PR TITLE
Add coordinate system to heating function

### PR DESCRIPTION
**Pull request by gassmoeller**
In regards to #5626.

I added support for different coordinate systems to the heating model plugin 'function'. It also looks like we did not have a test for that plugin in the first place, so I added tests for cartesian, depth, and spherical coordinate systems.

@alarshi This is approximately what I imagined in #5626, it may need to look different for the initial topograpy, but not much (e.g. depth coordinate system may not make sense).

### CODE REVIEW

File: tests/heating_function_depth.prm [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# A test for the function heating plugin.
+# The heating function is given in terms of depth.
```
Comment (jdannberg) [inline tests/heating_function_depth.prm:2]: If these tests are all the same, with the only difference being the coordinate system, can you include the one that uses the cartesian coordinates and only keep the changes in this file?

File: tests/heating_function_spherical.prm [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# A test for the function heating plugin.
+# The heating function is given in spherical coordinates.
```
Comment (jdannberg) [inline tests/heating_function_spherical.prm:2]: Same here.

**Review by jdannberg — APPROVED** [review]
Looks good to me, I just have some minor comments.

**Comment by gassmoeller** [comment]
I think I addressed all comments.

**Pull request merged**